File: src/Handlers/logger.ts
import chalk from "chalk";
import boxen from "boxen";
import type { StartupData } from "../types.js";

export const startupReport = (data: StartupData): void => {
  const lines = [
    `Slash Commands: ${data.slash}`,
    `Prefix Commands: ${data.prefix}`,
    `Events Loaded: ${data.events}`,
    `Models Loaded: ${data.models}`,
    `AntiCrash: ${data.anticrash ? "Active" : "Inactive"}`,
    `MongoDB: ${data.mongo ? "Connected" : "Disconnected"}`,
  ];

  const report = lines
    .map((l) => {
      if (l.includes("Disconnected") || l.includes("Inactive")) {
        return `  ❌ ${chalk.red(l)}`;
      }
      return `  ✅ ${chalk.green(l)}`;
    })
    .join("\n");

  const boxed = boxen(report, {
    title: chalk.blue.bold(data.name || "DISCORD HANDLER"),
    titleAlignment: "center",
    padding: 1,
    margin: 1,
    borderStyle: "round",
    borderColor: "cyan",
  });

  console.log(boxed);
  console.log(
    chalk.magentaBright(
      `[ ${new Date().toLocaleString("en-GB", { timeZone: "Asia/Riyadh" })} ]`
    ),
    chalk.green("Bot is now online and fully operational.")
  );
};

File: go/handlers/logger.go
package handlers

import (
	"fmt"
	"time"

	"github.com/fatih/color"
)

type StartupData struct {
	Name         string
	SlashCount   int
	PrefixCount  int
	EventsCount  int
	ModelsCount  int
	MongoStatus  bool
}

func StartupReport(data StartupData) {
	fmt.Println()
	color.Cyan("╔══════════════════════════════════╗")
	color.Cyan("║     Discord Handler - Go         ║")
	color.Cyan("╚══════════════════════════════════╝")
	fmt.Println()

	lines := []struct {
		label string
		ok    bool
	}{
		{fmt.Sprintf("Slash Commands: %d", data.SlashCount), true},
		{fmt.Sprintf("Prefix Commands: %d", data.PrefixCount), true},
		{fmt.Sprintf("Events Loaded: %d", data.EventsCount), true},
		{fmt.Sprintf("Models Loaded: %d", data.ModelsCount), true},
		{"AntiCrash: Active", true},
		{fmt.Sprintf("MongoDB: Connected = %v", data.MongoStatus), data.MongoStatus},
	}

	for _, l := range lines {
		if l.ok {
			color.Green("  ✅ %s", l.label)
		} else {
			color.Red("  ❌ %s", l.label)
		}
	}

	fmt.Println()
	color.Magenta("[ %s ] Bot is now online and fully operational.", time.Now().Format("02/01/2006 15:04:05"))
}
